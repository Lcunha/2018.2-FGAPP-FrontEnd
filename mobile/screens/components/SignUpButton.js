import styles from '../tab_navigator/vendas/styles';
import React from 'react';
import {Text} from 'react-native';
import {Button} from 'native-base';

class SignUpButton extends React.Component{
  render(){
    return(
        <Button
        bordered light block
        onPress={() => this.props.onPress()}
        color='transparent'
        >
          <Text style={styles.signUpBtn}>CADASTRAR</Text>
        </Button>
    );
  }
}

export default SignUpButton;
