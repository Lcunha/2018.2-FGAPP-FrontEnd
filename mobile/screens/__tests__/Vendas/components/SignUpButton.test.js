import React from 'react';
import SignUpButton from '../../../components/SignUpButton';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';

Enzyme.configure({adapter: new Adapter()});

  it('renders correctly', () => {
    const tree = renderer.create(<SignUpButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Testing onPress', () => {
    const onPressMock = jest.fn();
    const wrapper = shallow(<SignUpButton onPress = {onPressMock}/>);
    const button = wrapper.find('Styled(Button)').at(0);
    button.simulate('press');
    expect(onPressMock).toBeCalled();
  })