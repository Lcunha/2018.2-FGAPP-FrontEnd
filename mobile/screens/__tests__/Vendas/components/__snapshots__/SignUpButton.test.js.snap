// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "transparent",
      "borderBottomWidth": 1,
      "borderColor": "#f4f4f4",
      "borderLeftWidth": 1,
      "borderRadius": 5,
      "borderRightWidth": 1,
      "borderTopWidth": 1,
      "borderWidth": 1,
      "elevation": null,
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
      "shadowColor": null,
      "shadowOffset": null,
      "shadowOpacity": null,
      "shadowRadius": null,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontWeight": "bold",
      }
    }
  >
    CADASTRAR
  </Text>
</View>
`;
